# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ModuleLoader.h
# Commit: c11cffaded9c
# Full Hash: c11cffaded9cfed53b04e5c7875278e9a9e55396
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 21:52:29
# Regressor Bug: 1742437
# File Overlap Count: 5
# Description:
#   Bug 1742437 - re-scope all generic module/script loader files under js/loader; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138133
# ==============================================================================

diff -r 0527d227c6b6 -r c11cffaded9c dom/script/ModuleLoader.h
--- a/dom/script/ModuleLoader.h	Fri Feb 18 12:45:52 2022 +0000
+++ b/dom/script/ModuleLoader.h	Fri Feb 18 12:45:53 2022 +0000
@@ -7,9 +7,9 @@
 #ifndef mozilla_dom_ModuleLoader_h
 #define mozilla_dom_ModuleLoader_h
 
-#include "ModuleLoaderBase.h"
 #include "mozilla/dom/ScriptLoadContext.h"
-#include "mozilla/dom/ScriptLoadRequest.h"
+#include "js/loader/ModuleLoaderBase.h"
+#include "js/loader/ScriptLoadRequest.h"
 #include "ScriptLoader.h"
 
 class nsIURI;
@@ -18,19 +18,24 @@
 
 class CompileOptions;
 
+namespace loader {
+
+class ModuleLoadRequest;
+
+}  // namespace loader
 }  // namespace JS
 
 namespace mozilla {
 namespace dom {
 
-class ModuleLoadRequest;
+class ScriptLoader;
 class SRIMetadata;
 
 //////////////////////////////////////////////////////////////
 // DOM Module loader implementation
 //////////////////////////////////////////////////////////////
 
-class ModuleLoader final : public ModuleLoaderBase {
+class ModuleLoader final : public JS::loader::ModuleLoaderBase {
  private:
   virtual ~ModuleLoader();
 